src/olympia/addons/serializers.py

```diff
@@ -535,15 +536,32 @@ def create(self, validated_data):
         return version
 
     def update(self, instance, validated_data):
+        user = self.context['request'].user
         custom_license = (
             validated_data.pop('license')
             if isinstance(validated_data.get('license'), dict)
             else None
         )
+        existing_maxs = {
+            app: appver.max for app, appver in instance.compatible_apps.items()
+        }
 
         instance = super().update(instance, validated_data)
         if 'compatible_apps' in validated_data:
             instance.set_compatible_apps(validated_data['compatible_apps'])
+            for app, appver in instance.compatible_apps.items():
+                if app not in existing_maxs or existing_maxs[app] != appver.max:
+                    ActivityLog.create(
+                        amo.LOG.MAX_APPVERSION_UPDATED,
+                        instance.addon,
+                        instance,
+                        user=user,
+                        details={
+                            'version': instance.version,
+                            'target': appver.version.version,
+                            'application': appver.application,
+                        },
+                    )
```

Comment:
fwiw, I don't like how the logging in the Version serializer has turned out - it's all over the place.  `source` has it's own logging because we send an email too; and `license` and the max version change needs to know information from both before the update and after 😞